base/mpfr.jl

```diff
@@ -765,12 +765,15 @@ function with_bigfloat_precision(f::Function, precision::Integer)
 end
 
 function string(x::BigFloat)
-    lng::Int32 = 127
+    # In general, the number of decimal places needed to read back the number exactly is
+    # ceil(log(10, 2^precision(x))) or ceil(precision(x) * log10(2))
```

Comment:
Doesn't this need to be 1 higher? see, for example http://www.exploringbinary.com/number-of-digits-required-for-round-trip-conversions/ and http://www.exploringbinary.com/the-inequality-that-governs-round-trip-conversions-a-partial-proof/